examples/megacli.cpp

```diff
@@ -3576,18 +3590,131 @@ void exec_get(autocomplete::ACState& s)
+                    {
+                        cout << "Identical file already exist. Skipping transfer of " << name << endl;
+                        return;
+                    }
+                }
+                else
+                {
+                    cout << "Can't upload file over the top of a folder with the same name: " << name << endl;
+                    return;
+                }
+            }
+            fa.reset();
+
+            AppFile* f = new AppFilePut(&localname, parent ? parent->nodehandle : UNDEF, targetuser.c_str());
+            *static_cast<FileFingerprint*>(f) = fp;
+            f->appxfer_it = appxferq[PUT].insert(appxferq[PUT].end(), f);
+            client->startxfer(PUT, f, committer);
+            total++;
+        }
+        else
+        {
+            cout << "Can't open file: " << name << endl;
+        }
+    }
+    if (type == FOLDERNODE && recursive)
+    {
+
+        if (previousNode)
+        {
+            if (previousNode->type == FILENODE)
+            {
+                cout << "Can't upload a folder over the top of a file with the same name: " << name << endl;
+                return;
+            }
+            else
+            {
+                // upload into existing folder with the same name
+                uploadLocalFolderContent(localname, previousNode);
+            }
+        }
+        else
+        {
+            auto nn = new NewNode[1];
+            client->putnodes_prepareOneFolder(nn, name);
+
+            gOnPutNodeTag[g_nextClientTag] = [localname](Node* parent) {
+                uploadLocalFolderContent(localname, parent);
+            };
+
+            client->reqtag = g_nextClientTag++;
+            client->putnodes(parent->nodehandle, nn, 1);
+            client->reqtag = 0;
+        }
+    }
+}
+
+
+string localpathToUtf8Leaf(const string& itemlocalname)
+{
+    string::size_type pos = 0, testpos = 0;
+    while (string::npos != (testpos = itemlocalname.find(client->fsaccess->localseparator, pos)))
+    {
+        pos = testpos + client->fsaccess->localseparator.size();
+    }
+
+    string leafNameLocal = itemlocalname.substr(pos);
+    string leafNameUtf8;
+    client->fsaccess->local2path(&leafNameLocal, &leafNameUtf8);
+    return leafNameUtf8;
+}
+
+void uploadLocalFolderContent(std::string localname, Node* cloudFolder)
+{
+    DirAccess* da = client->fsaccess->newdiraccess();
+
+    if (da->dopen(&localname, NULL, false))
+    {
+        DBTableTransactionCommitter committer(client->tctable);
+
+        nodetype_t type;
+        string itemlocalleafname;
+        while (da->dnext(NULL, &itemlocalleafname, true, &type))
+        {
+            string leafNameUtf8 = localpathToUtf8Leaf(itemlocalleafname);
+
+            if (gVerboseMode)
+            {
+                cout << "Queueing " << leafNameUtf8 << "..." << endl;
+            }
+            int total = 0;
```

Comment:
I think `total` is unused

Reply:
good point - I've adjusted it so we do use the count, and output a message about more nodes queued for upload (in verbose mode).  thanks